tests/test_request_fingerprinter.py

```diff
@@ -0,0 +1,243 @@
+import sys
+from unittest.mock import patch
+
+import pytest
+from packaging.version import Version
+from scrapy import __version__ as SCRAPY_VERSION
+
+if Version(SCRAPY_VERSION) < Version("2.7"):
+    pytest.skip("Skipping tests for Scrapy < 2.7", allow_module_level=True)
+
+from importlib.metadata import version as package_version
+
+from scrapy import Request, Spider
+from web_poet import ItemPage, WebPage
+
+from scrapy_poet import ScrapyPoetRequestFingerprinter
+
+from . import get_crawler as _get_crawler
+
+ANDI_VERSION = Version(package_version("andi"))
+
+SETTINGS = {
+    "DOWNLOADER_MIDDLEWARES": {
+        "scrapy_poet.InjectionMiddleware": 543,
+    },
+    "REQUEST_FINGERPRINTER_CLASS": ScrapyPoetRequestFingerprinter,
+}
+
+
+def get_crawler(spider_cls=None, settings=None):
+    settings = SETTINGS if settings is None else settings
+    kwargs = {}
+    if spider_cls is not None:
+        kwargs["spider_cls"] = spider_cls
+    return _get_crawler(settings=settings, **kwargs)
+
+
```

Comment:
Let's also add a test with different URLs to get some smoke when refactoring in the future, just in case.

Reply:
Let's also add one with `DummyResponse` as well.

Reply:
Hmm… Regarding `DummyResponse`, I 100% agree on having the test, but I wonder what the outcome should be. Right now the typing of the response is ignored, and I think that can be fine, but it could also be argued that `DummyResponse` should generate a different fingerprint (since in some cases that could result in different underlying requests).

Reply:
On a related note, I wonder if we should ignore request inputs (e.g. RequestUrl, RequestBody, RequestHeaders) for fingerprinting, given they cannot alter the result.

Reply:
>  but I wonder what the outcome should be. Right now the typing of the response is ignored, and I think that can be fine

+1 on this. 

> but it could also be argued that DummyResponse should generate a different fingerprint (since in some cases that could result in different underlying requests).

The resulting different fp would pretty much be based on the other injected parameters anyways, so it's okay to be ignored.

>  I wonder if we should ignore request inputs (e.g. RequestUrl, RequestBody, RequestHeaders) for fingerprinting, given they cannot alter the result.

Hmm not sure about this, as there might be changes in the future that affects them. For example, `typing.Annotated` might be introduced to them.

Reply:
> The resulting different fp would pretty much be based on the other injected parameters anyways, so it's okay to be ignored.

I‘m thinking of:

```
def parse(self, response, request_url: RequestUrl)
def parse(self, response: DummyResponse, request_url: RequestUrl)
```

It is an unrealistic scenario, though, given that the use of `DummyResponse` is meant for cases where you get a response differently (e.g. BrowserHtml), scenarios where you don’t get a response at all seem impractical, so yes, I think you are right.

Reply:
> Hmm not sure about this, as there might be changes in the future that affects them. For example, typing.Annotated might be introduced to them.

What about [ignoring unannotated request and utility page inputs](https://github.com/scrapinghub/scrapy-poet/pull/172/commits/a611fd2714f60135af8299d184697dda1f9c3ac0) for now, and deciding what to do with their annotated versions later on (e.g. after merging https://github.com/scrapinghub/scrapy-poet/pull/169)?

Reply:
https://github.com/scrapinghub/scrapy-poet/pull/169 is merged, so later became now :sweat_smile:

Reply:
Oh, I had already added a test for annotated dependencies, never mind.